packages/avatar-group/src/vaadin-avatar-group.js

```diff
@@ -403,10 +431,41 @@ class AvatarGroup extends ResizeMixin(ElementMixin(ThemableMixin(PolymerElement)
   }
 
   /** @private */
-  __computeItems(arr, itemsInView, maxItemsVisible) {
+  __updateAvatars(arr, itemsInView, maxItemsVisible) {
     const items = arr.base || [];
     const limit = this.__getLimit(items.length, itemsInView, maxItemsVisible);
-    return limit ? items.slice(0, limit) : items;
+
+    const newItems = limit ? items.slice(0, limit) : items;
+    const oldItems = this.__oldAvatars || [];
+
+    if (newItems.length || oldItems.length) {
+      const removed = oldItems.filter((item) => !newItems.includes(item));
+      const added = [...newItems];
+
+      this._avatars.forEach((avatar) => {
+        const item = avatar._item;
+        if (removed.includes(item)) {
+          avatar.remove();
+        } else if (added.includes(item)) {
+          added.splice(added.indexOf(item), 1);
+        }
+      });
+
+      this.__addAvatars(added, newItems);
+    }
+
+    this._avatars = [...this.querySelectorAll('vaadin-avatar')];
+    this.__oldAvatars = newItems;
```

Comment:
Naming could be confusing here, as `this.__oldAvatars` contains items, not avatar element instances.

Reply:
Renamed to `__oldAvatarItems`.